cmake/add_cpp_test.cmake

```diff
@@ -41,13 +72,13 @@ function(add_cpp_test TEST_NAME)
   endif()
 
   # Add as a dependency of given targets
-  foreach(dependency_of ${ARGS_DEPENDENCY_OF})
+  foreach(dependency_of IN LISTS ARGS_DEPENDENCY_OF)
     if (NOT TARGET ${dependency_of}_tests)
       add_custom_target(${dependency_of}_tests)
       add_dependencies(all_tests ${dependency_of}_tests)
-    endif(NOT TARGET ${dependency_of}_tests)
+    endif()
```

Comment:
Is it recommended that we not put stuff in the endif()?

Reply:
It probably was required in some early CMake versions, but it's been optional for the longest time. Proper indentation already makes it easy to tell blocks apart and if the expressions don't match that's an error, it's just better not have anything in the closing command call.

Reply:
I felt like the end bracket stuff made for nice commentary, so you could tell what was being closed when the opening line is far away. Guess I am just wondering if there is a common style (if you don't know, it's cool).